sqlglot/generator.py

```diff
@@ -3112,6 +3112,9 @@ def toarray_sql(self, expression: exp.ToArray) -> str:
         cond_for_null = arg.is_(exp.null())
         return self.sql(exp.func("IF", cond_for_null, exp.null(), exp.Array(expressions=[arg])))
 
+    def tsordstotime_sql(self, expression: exp.TsOrDsToTime) -> str:
```

Comment:
what's this? this kills round trip

Reply:
how? this is done to avoid overriding everywhere with a cast needlessly, if you need an identity roundtrip then you override the transform entry like I did for BQ right?

Reply:
that’s what we do for tsordstodate, if you’re going to do that you should refactor that as well

Reply:
I see, let me take another look and I'll refactor if needed then

Reply:
tsordstodate could be cleaned up if we had this just need to make sure we don’t have extraneous casts

Reply:
take a look here @tobymao https://github.com/tobymao/sqlglot/pull/2752/commits/bcf2b912fdff69a1b86366c2190ab231ff07e8eb